server/api/locations/locations.api.js

```diff
@@ -0,0 +1,118 @@
+const serialize = require('express-serializer');
+
+const { route } = require('../utils/api');
+const { validateRequestBody } = require('../utils/validation');
+const { point: validateGeoJsonPoint } = require('../validators/geojson');
+const policy = require('./locations.policy');
+
+exports.create = route(async (req, res) => {
+
+  await validateLocation(req);
+
+  const location = policy.parse(req.body);
+  await location.save();
+
+  res.status(201).send(await serialize(req, location, policy));
+});
+
+
+function validateLocation(req) {
+  return validateRequestBody(req, function() {
+    return this.parallel(
+      this.validate(
+        this.json('/name'),
+        this.required(),
+        this.type('string'),
+        this.notBlank(),
+        this.string(1, 150)
+      ),
+      this.validate(
+        this.json('/shortName'),
+        this.while(this.isSet()),
+        this.type('string'),
+        this.notBlank(),
+        this.string(1, 30)
+      ),
+      this.validate(
+        this.json('/description'),
+        this.required(),
+        this.type('string'),
+        this.notBlank(),
+        this.string(1, 2000)
```

Comment:
Why limit to 2000 ?
This is quite a long string and will probably be enough for all the descriptions. Just wondering.

Reply:
Not sure what we intend to put in the description. I took a lorem ipsum and generated 2000 characters. It looked like 3-4 solid paragraphs, so I thought that should be good for whatever we need.